php/src/Google/Protobuf/Internal/Message.php

```diff
@@ -1035,6 +1074,24 @@ protected function mergeFromJsonArray($array)
             }
             return;
         }
+        if ($is_json) {
+            $this->mergeFromArrayJsonImpl($array);
+        } else {
+            // Just call the setters for the field names
+            foreach ($array as $key => $value) {
+                $field = $this->desc->getFieldByJsonName($key)
+                    ?: $this->desc->getFieldByName($key);
+                if (is_null($field)) {
```

Comment:
This is if the `$field` object is null though. I am not sure in what case this could arise, but if it did we'd get a PHP Fatal error on the call to `$field->getSetter()` in the next line.

Reply:
oh, sorry, I misunderstood previously. Shall we also throw an exception in this case?